java/src/jmri/util/HelpUtil.java

```diff
@@ -71,115 +64,93 @@ public static JMenu makeHelpMenu(String ref, boolean direct) {
     }
 
     public static JMenuItem makeHelpMenuItem(String ref) {
-        if (!initOK()) {
-            return null;  // initialization failed
-        }
         JMenuItem menuItem = new JMenuItem(Bundle.getMessage("MenuItemWindowHelp"));
-        globalHelpBroker.enableHelpOnButton(menuItem, ref, null);
-
-        // start help to see what happend
-        log.debug("help: {}:{}:{}", globalHelpSet.getHomeID(), globalHelpSet.getTitle(), globalHelpSet.getHelpSetURL());
+        
+        menuItem.addActionListener((ignore) -> {
+            displayHelpRef(ref);
+        });
 
         return menuItem;
     }
 
     public static void addHelpToComponent(java.awt.Component component, String ref) {
-        if (globalHelpBroker != null) {
-            globalHelpBroker.enableHelpOnButton(component, ref, null);
-            log.debug("Help added for {}", ref);
+        enableHelpOnButton(component, ref);
+    }
+
+    // https://coderanch.com/how-to/javadoc/javahelp-2.0_05/javax/help/HelpBroker.html#enableHelpOnButton(java.awt.Component,%20java.lang.String,%20javax.help.HelpSet)
+    public static void enableHelpOnButton(java.awt.Component comp, String id) {
+        if (comp instanceof javax.swing.AbstractButton) {
+            ((javax.swing.AbstractButton)comp).addActionListener((ignore) -> {
+                displayHelpRef(id);
+            });
+        } else if (comp instanceof java.awt.Button) {
+            ((java.awt.Button)comp).addActionListener((ignore) -> {
+                displayHelpRef(id);
+            });
         } else {
-            log.debug("globalHelpBroker is null");
+            throw new IllegalArgumentException("comp is not a javax.swing.AbstractButton or a java.awt.Button");
         }
     }
 
     public static void displayHelpRef(String ref) {
-        if (globalHelpBroker == null) {
-            log.debug("can't display {} help page because help system reference is null", ref);
-            return;
+        HelpUtilPreferences preferences = InstanceManager.getDefault(HelpUtilPreferences.class);
+        String file = "help/en/" + ref.replace(".", "/") + ".shtml";
```

Comment:
This code needs to handle references like `package.jmri.jmrit.ctc.CTC_menuCfgDef`.

The link in index.html will be:

```
                                                    <li>
                                                        <a onclick="
                        openLink('../package/jmri/jmrit/ctc/CTC.shtml#menuCfgDef')
                      ">Defaults </a>
                                                        <ul></ul>
                                                    </li>
```
The underscore character splits between the url and the anchor.